components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/oidcdiscovery/OIDCDiscoveryEndpoint.java

```diff
@@ -98,7 +99,7 @@ private Response getResponse(HttpServletRequest request, String tenant) {
         String response;
         OIDCProcessor processor = EndpointUtil.getOIDCService();
         try {
-            OIDProviderResponseBuilder responseBuilder = new OIDProviderJSONResponseBuilder();
+            OIDProviderResponseBuilder responseBuilder = getOidProviderResponseBuilder();
```

Comment:
Can we consider renaming the class name OIDProviderResponseBuilder?
OIDCProviderDiscoveryResponseBuilder or OpenIDConnectProviderDiscoveryResponseBuilder 

or something along the lines that would give the purpose of the extension or interface.

Reply:
For backward compatibility we will keep the name as it is. We will improve the Java Docs of this interface to convey the actual purpose of the interface.